### PR TITLE
chore setup redux

### PR DESCRIPTION
**Pull request by nicele08**
**What does this PR do?**
> - setting up redux

**Description of Task to be completed**
> - add counter
> - create actions
> - create reducers

**How should this be manually tested?**
> - git clone to `https://github.com/atlp-rwanda/elite-bn-frontend.git`
> - checkout to `ch-redux-setup`
> - `npm install` to install node_modules
> - `npm start ` to start server
> - open http://localhost:8080/ to view it in the browser.

**What are the relevant Trello story?**
> - https://trello.com/c/SuAj3WDt

**Screenshot**
> - ch-setup-redux

### CODE REVIEW

File: src/store/index.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import resultReducer from './reducers/result';
+
+const rootReducer = combineReducers({
+  ctr: counterReducer,
+  res: resultReducer,
+});
+
+const store = createStore(rootReducer);
+
+export default store;
```
Comment (b0nbon1) [inline src/store/index.js]: please also setup redux middlewares i.e redux thunk and redux devtools

File: src/store/reducers/counter.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import * as actionTypes from '../../constants/actions';
+import * as initials from '../initialState';
```
Comment (b0nbon1) [inline src/store/reducers/counter.js]: please destructure your import and import every element indivudually, eg. `import { INCREMENT } from '...'`.

File: src/components/Counter/CounterControl.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+const counterControl = (props) => (
+  <div className="w-36 px-8 py-6 border-yellow-400 shadow-sm font-bold cursor-pointer text-center inline-block bg-gray-400 ml-4" onClick={props.clicked}>
```
Comment (b0nbon1) [inline src/components/Counter/CounterControl.js]: this line is too long please break it down to multiple lines. You can add this setting to your `.prettierrc` file. 1 line should have maximum length of 100 characters for neat code and also install prettier as devdepednancy

Comment (b0nbon1) [reply src/components/Counter/CounterControl.js]:
```js
{
  "printWidth": 100,
  "trailingComma": "all",
  "semi": true,
  "jsxSingleQuote": true,
  "singleQuote": true,
  "tabWidth": 2,
  "arrowParens": "avoid"
}
```

File: src/Routes/index.js [hunk 4]
```diff
@@ -1,10 +1,12 @@
 import React from 'react';
 import { Switch, Route } from 'react-router-dom';
 import Home from '../components/Home';
+import Counter from '../containers/Counter/Counter';
 
 function index() {
   return (
     <Switch>
+      <Route exact path="/" component={Counter} />
```
Comment (b0nbon1) [inline src/Routes/index.js]: I know it's an example to show it is working, but could you use different route pathname for them

**Review by b0nbon1 — CHANGES_REQUESTED** [review]

File: .eslintrc.js [hunk 5]
```diff
@@ -19,5 +22,22 @@ module.exports = {
...
   rules: {
     'react/jsx-filename-extension': 'off',
+    'linebreak-style': [0, 'windows'],
+    'import/no-named-as-default': 0,
+    'import/no-named-as-default-member': 0,
+    'default-case': 'off',
+    'react/prefer-stateless-function': 0,
+    'react/state-in-constructor': 0,
+    'react/destructuring-assignment': 0,
+    'prefer-destructuring': 'off',
```
Comment (pextech) [inline .eslintrc.js]: why is this off?

File: src/Routes/index.js [hunk 6]
```diff
@@ -1,10 +1,12 @@
 import React from 'react';
 import { Switch, Route } from 'react-router-dom';
 import Home from '../components/Home';
+import Counter from '../containers/Counter/Counter';
 
 function index() {
   return (
     <Switch>
+      <Route exact path="/counter" component={Counter} />
```
Comment (pextech) [inline src/Routes/index.js:9]: i think we do need one exact path

**Review by pextech — CHANGES_REQUESTED** [review]

File: src/Routes/index.js (hunk 6 above)
Comment (nicele08) [reply src/Routes/index.js:9]: `exact` disables the partial matching for a route. 
Here will return the route if only the path is exact match to this URL

File: .eslintrc.js (hunk 5 above)
Comment (nicele08) [reply .eslintrc.js]: Here I was getting errors from eslint when I created the component.
May be we have to configure eslint properly before we continue to work on other features.

**Review by NIRoberto — APPROVED** [review]
LGTM

**Review by 444root — APPROVED** [review]

**Review by b0nbon1 — APPROVED** [review]

**Review by iradukundairenee — APPROVED** [review]

**Review by pextech — APPROVED** [review]

**Pull request merged**